src/diffpy/snmf/snmf_class.py

```diff
@@ -106,66 +106,72 @@ def __init__(
             objective function to allow without terminating the optimization. Note that
             a minimum of 20 updates are run before this parameter is checked.
         n_components : int
-            The number of components to extract from MM. Must be provided when and only when
+            The number of components to extract from source_matrix. Must be provided when and only when
             Y0 is not provided.
         random_state : int
             The seed for the initial guesses at the matrices (A, X, and Y) created by
             the decomposition.
         """
 
-        self.MM = MM
+        self.source_matrix = source_matrix
         self.rho = rho
         self.eta = eta
         # Capture matrix dimensions
-        self._N, self._M = MM.shape
+        self._signal_len, self._num_conditions = source_matrix.shape
         self.num_updates = 0
         self._rng = np.random.default_rng(random_state)
 
         # Enforce exclusive specification of n_components or Y0
-        if (n_components is None) == (Y0 is not None):
-            raise ValueError("Must provide exactly one of Y0 or n_components, but not both.")
-
-        # Initialize Y0 and determine number of components
-        if Y0 is None:
-            self._K = n_components
-            self.Y = self._rng.beta(a=2.5, b=1.5, size=(self._K, self._M))
+        if (n_components is None and init_weights is None) or (
+            n_components is not None and init_weights is not None
+        ):
+            raise ValueError("Must provide exactly one of init_weights or n_components, but not both.")
```

Comment:
please make all error messages of hte form "what went wrong.  What to do about it."  Yours is almost that, but it could be done a bit more cleanly.